scripts/dnp3-gen/dnp3-gen.py

```diff
@@ -0,0 +1,707 @@
+#! /usr/bin/env python
+#
+# Copyright (C) 2015 Open Information Security Foundation
+#
+# You can copy, redistribute or modify this Program under the terms of
+# the GNU General Public License version 2 as published by the Free
+# Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# version 2 along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+# This script generates DNP3 related source code based on definitions
+# of DNP3 objects (currently the object structs).
+
+from __future__ import print_function
+
+import sys
+import re
+from cStringIO import StringIO
+import yaml
+import types
+
+import jinja2
+
+IN_PLACE_START = "/* START GENERATED CODE */"
+IN_PLACE_END = "/* END GENERATED CODE */"
+
+util_lua_dnp3_objects_c_template = """/* Copyright (C) 2015 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * DO NOT EDIT. THIS FILE IS AUTO-GENERATED.
```

Comment:
Ok, all generated files, or segments have a comment, right now it looks like:

/* Code generated by:
 *     ./scripts/dnp3-gen/dnp3-gen.py
 */